### ISSUE TITLE
regenerate leafcutter container

### ISSUE THREAD

**Issue by kerimoff** [eQTL-Catalogue/rnaseq#3]
Current docker container for leafcutter related processes are build by Karl Vaba and are not under eQTLCatalogue organisation.

Re-build them and push to the correct repository

**Comment by yangchuhua** [comment]
Hi,

Thanks for your efforts on this project.
I got the following error and I wonder if I need to modify the main.nf file:
```
Channel `bam_count_exons` has been used twice as an output by process `sort_by_name_BAM` and another operator

 -- Check script '/home/yang/software/rnaseq/main.nf' at line: 818 or see '.nextflow.log' file for more details`
```

When I ran this process by the following codes: 
```
nextflow run ~/software/rnaseq \
        -profile docker \
	--readPathsFile ${SampleSheet_dir}/readPathsFile_AOR_SE.tsv \
        --singleEnd \
        --unstranded \
        --star_index ${referenceDir}/hg38_genocode_v26_primary_star_index_sjdbOverhang149 \
        --aligner 'star' \
        --saveReference \
        --saveTrimmed \
        --saveAlignedIntermediates \
        --fasta ${referenceDir}/GENCODE_GRCh38_V26/GENCODE_GRCh38_V26/GRCh38.p10.genome.fa \
        --gff ${referenceDir}/GENCODE_GRCh38_V26/GENCODE_GRCh38_V26/gencode.v26.chr_patch_hapl_scaff.annotation.gff3 \
        --bed12 ${referenceDir}/GENCODE_GRCh38_V26/GENCODE_GRCh38_V26/genes.bed \
	--outdir ${OUTDIR} \
        --work-dir ${OUTDIR}
```

And the .nextflow.log shows the following error as before:
```
Feb-15 13:38:51.437 [main] ERROR nextflow.cli.Launcher - @unknown
nextflow.dag.MultipleOutputChannelException: Channel `bam_count_exons` has been used twice as an output by process `sort_by_name_BAM` and another operator
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:72)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:295)
        at nextflow.dag.DAG.outbound(DAG.groovy:233)
        at sun.reflect.GeneratedMethodAccessor30.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
        at nextflow.dag.DAG.addVertex(DAG.groovy:149)
        at sun.reflect.GeneratedMethodAccessor61.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
        at nextflow.dag.DAG.addProcessNode(DAG.groovy:103)
        at nextflow.dag.NodeMarker.addProcessNode(NodeMarker.groovy:52)
        at nextflow.dag.NodeMarker$addProcessNode$0.call(Unknown Source)
        at nextflow.processor.TaskProcessor.createOperator(TaskProcessor.groovy:530)
        at nextflow.processor.TaskProcessor.run(TaskProcessor.groovy:431)
        at nextflow.processor.TaskProcessor$run.call(Unknown Source)
        at nextflow.script.BaseScript.process(BaseScript.groovy:114)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:194)
        at Script_35056600.runScript(Script_35056600:818)
        at nextflow.script.BaseScript.runDsl1(BaseScript.groovy:163)
        at nextflow.script.BaseScript.run(BaseScript.groovy:200)
        at nextflow.script.ScriptParser.runScript(ScriptParser.groovy:221)
        at nextflow.script.ScriptRunner.run(ScriptRunner.groovy:212)
        at nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:120)
        at nextflow.cli.CmdRun.run(CmdRun.groovy:308)
        at nextflow.cli.Launcher.run(Launcher.groovy:480)
        at nextflow.cli.Launcher.main(Launcher.groovy:639)
```
Best ,
Chuhua

**Comment by kauralasoo** [comment]
Dear Chuhua,

Could you please submit your error as separate issue instead of commenting here?

I think that the main issue is that you are trying to use the STAR aligner that is currently not properly supported in our workflow (but the documentation does not reflect that).

We are in the middle of a major re-write of the workflow right now and plan to update the documentation once that's done.

What is bigger aim that you are trying to achieve? If you just need a gene expression matrix, I would use the nf-core/rnaseq workflow instead of ours. 

Best wishes, 
Kaur

**Comment by yangchuhua** [comment]
Hi Kaur,


Yes, you are right. I have changed to 'HISAT2'. And it works well.


Thanks for your reply.


Best,

Chuhua


-----------------------------

Chuhua Yang


Deutsches Herzzentrum München

Technische Universität München

Lazarettstraße 36

80636 München

Telefon: +34-610918421

________________________________

**Comment by kerimoff** [comment]
This issue has been solved with commit d1685c2ca7438d524425f7b0d43e3dfdab982a3c
